docs/callbacks.rst

```diff
@@ -0,0 +1,121 @@
+.. role:: python(code)
+          :language: python
+
+.. _callbacks:
+============================================================
+Callbacks
+============================================================
+
+LBANN has numerous callbacks that can be used to collect
+data about an experiment, such as scalars, metrics, weights,
+memory usage, images, etc. The documentation of many of these
+is pending; see the :ref:`list of Available
+Callbacks<available-callbacks>` for the documented ones.
+
+The callbacks are set to execute at various times, and can be
+used for reporting, debugging, and performing advanced training
+techniques.
+
+For a complete listing of callbacks and details about their
+functionality, please see :ref:`Available
+Callbacks<available-callbacks>`.
+
+.. _using-callbacks:
+------------------------------------------------
+Using Callbacks
+------------------------------------------------
+
+Callbacks are used by adding them to the python front end with the
+appropriate arguments and passing them as a list into the model.
+For example, the callbacks timer, print_statistics, and save model
+could be included with the following:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Python Front End
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: python
+
+   timer = lbann.CallbackTimer()
+   print_stats = lbann.CallbackPrintStatistics(
+                 batch_interval=5)
+   save_model = lbann.CallbackSaveModel(
+                dir=".",
+                disable_save_after_training=True)
+
+   callbacks = [timer,
+                print_stats,
+                save_model]
+
+   model = lbann.Model(num_epochs,
+                       layers,
+                       objective_function,
+                       metrics,
+                       callbacks)
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Profobuf (Advanced)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+   callback {
+     timer {
+     }
+     print_atatistics {
+       batch_interval: 5
+     }
+     save_model {
+       dir: "."
+       disable_save_after_training: true
+     }
+   }
+
+.. _available-callbacks:
+
+------------------------------------------------
+Available Callbacks
+------------------------------------------------
+
+.. toctree::
+   :maxdepth: 1
+
+   Check dataset <callbacks/check_dataset>
+   Check gradients <callbacks/check_gradients>
+   Check init <callbacks/check_init>
+   Check metric <callbacks/check_metric>
+   Check nan <callbacks/check_nan>
```

Comment:
_Check nan_ and _check small_ should be more descriptive e.g., Check nan in activation values